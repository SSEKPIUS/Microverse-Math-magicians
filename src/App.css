.App {
  text-align: center;
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.calc.grid-container {
  display: grid;
  grid-template-columns: auto auto auto auto;
  gap: 1px;
  background-color: #6f7275;
  padding: 1px;
}

.calc.grid-container > div {
  background-color: rgba(255, 255, 255, 0.8);
  text-align: center;
  padding: 20px;
  font-size: 30px;
  width: 2.5em;
  color: black;
  cursor: pointer;
}

.calc.grid-container .top {
  background-color: #6f7275;
  text-align: right;
  width: auto !important;
  color: white !important;
  cursor: default !important;
}

.calc.grid-container .right {
  background-color: #ed8d05;
}

.item1 {
  grid-column-start: 1;
  grid-column-end: 5;
}

.item18 {
  grid-column-start: 1;
  grid-column-end: 3;
  width: auto !important;
}
