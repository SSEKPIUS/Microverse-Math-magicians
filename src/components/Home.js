import React from 'react';

const Home = () => (
  <>
    <h2>Welcome to our page!</h2>
    <p>
      Labore lorem ipsum gubergren ipsum vero no sed amet ipsum dolor. Vero aliquyam
      tempor diam sit accusam. Voluptua ea sit et et justo tempor magna.
      Ipsum et vero clita invidunt sed diam tempor sit, lorem est sit no et
      elitr duo consetetur ipsum dolores. Eos et duo kasd kasd clita ipsum. Est ipsum
      no dolore amet. Et eos dolor sit diam sed eos lorem no, amet tempor ea
      consetetur clita amet at vero, dolores amet gubergren rebum duo, et takimata nonumy
      justo et accusam consetetur. Elitr est invidunt dolor et labore magna
      aliquyam dolor. Eos lorem sit et sed est. Aliquyam amet et duo magna duo
      sit, vero ipsum ipsum lorem elitr sit et sed, erat sit erat vero gubergren
      labore. Labore dolore diam est accusam. Erat consetetur tempor lorem
      sanctus sit, et diam sea ut dolor dolor, dolor dolor amet erat ea est. Accusam ut sed takimata
    </p>
    <p>
      Nun versuch faßt zu euch die kommt blick beifall. Glück lauf nun schwebet
      einer stunden, und gedränge beifall an gut. Bilder in und kommt im des sich
      der drängt. Ihr halbverklungnen entwöhntes mit lauf mein freundschaft trüben
      weich kommt, irren äolsharfe gleich lied erschüttert wahn mich. Das jenem
      ihr jugendlich geisterreich und alten der wiederholt. Zerstoben tränen
      manche der lied, zerstoben euch sie mit die, die des nennt die macht
      gezeigt die liebe was welt. Sage wahn ach mild irren. Gesänge meinem irren
      euch und sich die und im. Gut äolsharfe mein wenn ertönt mit die. Ich lied faßt noch.
    </p>
  </>
);

export default Home;
