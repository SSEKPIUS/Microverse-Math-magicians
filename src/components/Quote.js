import React from 'react';

const Quote = () => (
  <div className="quote">
    <p>
      Mathematics is not about numbers, equations, computations, or algprithms. it is
      about understanding.  - william Paul Thurstan.
    </p>
  </div>
);

export default Quote;
