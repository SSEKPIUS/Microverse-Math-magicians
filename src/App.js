import React from 'react';
import { Route, Routes } from 'react-router-dom';
import Calculate from './components/Calculate';
import Home from './components/Home';
import Quote from './components/Quote';
import Error from './components/Error';
import Nav from './components/Nav';

const App = () => (
  <>
    <div className="header">
      <div className="container">
        <Nav />
      </div>
    </div>
    <div className="body">
      <div className="container">
        <Routes>
          <Route path="/" element={<Home />} exact />
          <Route path="/calculator" element={<Calculate />} />
          <Route path="/quote" element={<Quote />} />
          <Route path="*" element={<Error />} />
        </Routes>
      </div>
    </div>
    <div className="footer">
      <div className="container">
        <span>Microverse Math magicians: full website @2022</span>
      </div>
    </div>
  </>
);
export default App;
