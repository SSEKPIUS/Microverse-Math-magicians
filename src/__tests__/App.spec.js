import React from 'react';

import Enzyme, { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import calculate from '../logic/calculate';
import operate from '../logic/operate';
import App from '../App';
import Error from '../components/Error';
import Home from '../components/Home';
import Key from '../components/key';
import Quote from '../components/Quote';

Enzyme.configure({ adapter: new Adapter() });
beforeEach(() => {
  jest.resetAllMocks();
});

describe('Math Magicians: tests', () => {
  describe('React components', () => {
    test('AP renders correctly enzyme', () => {
      const wrapper = shallow(<App />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('Error renders correctly enzyme', () => {
      const wrapper = shallow(<Error />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('Home renders correctly enzyme', () => {
      const wrapper = shallow(<Home />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('Key renders correctly enzyme', () => {
      const wrapper = shallow(<Key />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('Quote renders correctly enzyme', () => {
      const wrapper = shallow(<Quote />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('operate.js test', () => {
    test('AC', () => {
      expect(
        calculate({ total: null, next: null, operation: null }, 'AC'),
      ).toMatchObject({ total: null, next: null, operation: null });
    });

    test('Addition', () => {
      expect(
        calculate({ total: 4, next: 2, operation: '+' }, '='),
      ).toMatchObject({ total: '6', next: null, operation: null });
    });

    test('Subtraction', () => {
      expect(
        calculate({ total: 4, next: 2, operation: '-' }, '='),
      ).toMatchObject({ total: '2', next: null, operation: null });
    });
  });

  describe('calculate.js test', () => {
    test('Addition', () => {
      expect(operate(4, 4, '+')).toBe('8');
    });

    test('Subtraction', () => {
      expect(operate(5, 4, '-')).toBe('1');
    });

    test('Multiplication', () => {
      expect(operate(5, 4, 'x')).toBe('20');
    });

    test('Division', () => {
      expect(operate(10, 2, 'รท')).toBe('5');
    });

    test('Modulo', () => {
      expect(operate(10, 2, '%')).toBe('0');
    });
  });
});
