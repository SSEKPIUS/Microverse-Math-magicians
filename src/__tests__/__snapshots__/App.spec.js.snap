// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Math Magicians: tests React components AP renders correctly enzyme 1`] = `
<Fragment>
  <div
    className="header"
  >
    <div
      className="container"
    >
      <Nav />
    </div>
  </div>
  <div
    className="body"
  >
    <div
      className="container"
    >
      <Routes>
        <Route
          element={<Home />}
          exact={true}
          path="/"
        />
        <Route
          element={<Calculate />}
          path="/calculator"
        />
        <Route
          element={<Quote />}
          path="/quote"
        />
        <Route
          element={<Error />}
          path="*"
        />
      </Routes>
    </div>
  </div>
  <div
    className="footer"
  >
    <div
      className="container"
    >
      <span>
        Microverse Math magicians: full website @2022
      </span>
    </div>
  </div>
</Fragment>
`;

exports[`Math Magicians: tests React components Error renders correctly enzyme 1`] = `
<Fragment>
  <div
    className="error"
  >
    <span>
      <span
        style={
          Object {
            "color": "red",
          }
        }
      >
        Error! 
      </span>
      Page not Found
    </span>
  </div>
</Fragment>
`;

exports[`Math Magicians: tests React components Home renders correctly enzyme 1`] = `
<Fragment>
  <h2
    style={
      Object {
        "marginBottom": "3em",
        "marginTop": "4em",
      }
    }
  >
    Welcome to our page!
  </h2>
  <p>
    Labore lorem ipsum gubergren ipsum vero no sed amet ipsum dolor. Vero aliquyam tempor diam sit accusam. Voluptua ea sit et et justo tempor magna. Ipsum et vero clita invidunt sed diam tempor sit, lorem est sit no et elitr duo consetetur ipsum dolores. Eos et duo kasd kasd clita ipsum. Est ipsum no dolore amet. Et eos dolor sit diam sed eos lorem no, amet tempor ea consetetur clita amet at vero, dolores amet gubergren rebum duo, et takimata nonumy justo et accusam consetetur. Elitr est invidunt dolor et labore magna aliquyam dolor. Eos lorem sit et sed est. Aliquyam amet et duo magna duo sit, vero ipsum ipsum lorem elitr sit et sed, erat sit erat vero gubergren labore. Labore dolore diam est accusam. Erat consetetur tempor lorem sanctus sit, et diam sea ut dolor dolor, dolor dolor amet erat ea est. Accusam ut sed takimata
  </p>
  <p>
    Nun versuch faßt zu euch die kommt blick beifall. Glück lauf nun schwebet einer stunden, und gedränge beifall an gut. Bilder in und kommt im des sich der drängt. Ihr halbverklungnen entwöhntes mit lauf mein freundschaft trüben weich kommt, irren äolsharfe gleich lied erschüttert wahn mich. Das jenem ihr jugendlich geisterreich und alten der wiederholt. Zerstoben tränen manche der lied, zerstoben euch sie mit die, die des nennt die macht gezeigt die liebe was welt. Sage wahn ach mild irren. Gesänge meinem irren euch und sich die und im. Gut äolsharfe mein wenn ertönt mit die. Ich lied faßt noch.
  </p>
</Fragment>
`;

exports[`Math Magicians: tests React components Key renders correctly enzyme 1`] = `
<div
  aria-hidden="true"
  onClick={[Function]}
/>
`;

exports[`Math Magicians: tests React components Quote renders correctly enzyme 1`] = `
<div
  className="quote"
>
  <p>
    Mathematics is not about numbers, equations, computations, or algprithms. it is about understanding.  - william Paul Thurstan.
  </p>
</div>
`;
